Fix missing reference in diff/incr backup.

dwsteele opened the pull request:
When loading prior manifests without the new reference list, the code failed to add the current backup to the reference list. Since the current backup can never be referenced, building references from the file list was not sufficient to generate a complete list.

pgstef reviewed (approved):
The PR seems to fix the mentioned issue:

```
$ pgbackrest version
pgBackRest 2.42

$ pgbackrest backup --type=incr --stanza=ro8pg --log-level-console=info
...
P00   INFO: last backup label = 20221125-123310F_20221125-123345I, version = 2.41
...
ASSERT: [025]: unable to find '20221125-123310F_20221125-123345I' in string list

$ pgbackrest version
pgBackRest 2.43dev

$ pgbackrest backup --type=incr --stanza=ro8pg --log-level-console=info
...
P00   INFO: last backup label = 20221125-123310F_20221125-123345I, version = 2.41
...
P00   INFO: new backup label = 20221125-123310F_20221125-124858I
...
P00   INFO: backup command end: completed successfully (4864ms)
```

dwsteele commented:
I made some modifications after thinking about this some more.

The main problem here was a bad test which I fixed in 28f660482aa5abb97850a9d7c89c2e5e9f23a4c9. This masked the issue and prevented it from being found. I fixed that separately and now it is clear in the test that the current label is missing.

I moved the code to add the current label so it will be a noop if there is nothing to do, i.e. the reference list was read from the manifest.

pgstef commented:
@dwsteele, looks good to me :+1:
